Chat App/View/Pub/PubChatRoomViewController.swift

```diff
@@ -0,0 +1,125 @@
+        messagesCollectionView.messagesLayoutDelegate = self
+        messagesCollectionView.messagesDisplayDelegate = self
+        
+        let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout
+        layout?.setMessageIncomingAvatarSize(.zero)
+        layout?.setMessageOutgoingAvatarSize(.zero)
+        layout?.setMessageOutgoingMessageBottomLabelAlignment(LabelAlignment(textAlignment: .right, textInsets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 8)))
+        layout?.setMessageIncomingMessageBottomLabelAlignment(LabelAlignment(textAlignment: .left, textInsets: UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 0) ))
+    }
+}
+
+extension PubChatRoomViewController: InputBarAccessoryViewDelegate {
+    
+    func inputBar(_ inputBar: InputBarAccessoryView, didPressSendButtonWith text: String) {
+         let message = Message(content: text)
+        // TODO: Message input and message data source will be filled in by viewmodel
+        inputBar.inputTextView.text = ""
+    }
+}
+
+extension PubChatRoomViewController: MessagesLayoutDelegate {
+    
+    func configureAvatarView(_ avatarView: AvatarView, for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) {
+        avatarView.isHidden = true
+    }
+  
+    func footerViewSize(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CGSize {
+        return CGSize(width: 0, height: 8)
+    }
+  
+    func heightForLocation(message: MessageType, at indexPath: IndexPath, with maxWidth: CGFloat, in messagesCollectionView: MessagesCollectionView) -> CGFloat {
+        return 0
+    }
+    
+    func messageBottomLabelAttributedText(for message: MessageType, at indexPath: IndexPath) -> NSAttributedString? {
+        let value = message.sender.displayName
+        let attributes = [NSAttributedString.Key.font: UIFont(name: "Helvetica", size: 10)! ]
+        let attributedString = NSAttributedString(string: value, attributes: attributes)
+        return attributedString
+    }
+    
+    func messageBottomLabelHeight(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CGFloat {
+        return 10
+    }
+}
+
+extension PubChatRoomViewController: MessagesDataSource {
+    
+    func currentSender() -> SenderType {
+        return Sender(senderId: "", displayName: "") // TODO: this data will be filled from user defaults
+    }
+    
+    func numberOfMessages(in messagesCollectionView: MessagesCollectionView) -> Int {
+      return messagesThread.count
+    }
+    
+    func messageForItem(at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageType {
+        return messagesThread[indexPath.section]
+    }
+    
+    func numberOfSections(in messagesCollectionView: MessagesCollectionView) -> Int {
+        if messagesThread.count == 0 {
```

Comment:
if the array is empty and u called `.count` it will return zero so no need to validate its count here

Reply:
Removed redundant validation.